File: PSET6/sentimental-readability/readability.py
from cs50 import get_string

t = get_string("Text: ")

le = 0
wo = 1
se = 0
n = len(t)

for i in range(n):
    if t[i].isalpha():
        le += 1

    if t[i] == ' ':
        wo += 1

    if t[i] in ['.', '?', '!']:
        se += 1

L = 100 * (le / wo)
S = 100 * (se / wo)

# formula
index = round(0.0588 * L - 0.296 * S - 15.8)

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: PSET2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>


int main(void)
{
    string text = get_string("Text: ");

    float le = 0;
    float wo = 1;
    float se = 0;
    float n = strlen(text);

    for (int i = 0; i < n; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            le++;
        }

        if (text[i] == ' ')
        {
            wo++;
        }

        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            se++;
        }
    }

    float L = 100 * (le / wo);

    float S = 100 * (se / wo);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

    return 0;
}
